src/components/TodoContainer.js

```diff
@@ -10,15 +10,23 @@ import TodoNavigation from "./TodoNavigation";
 const TodoContainer = ({ tableName }) => {
   const [todoList, setTodoList] = useState([]);
   const [isLoading, setIsLoading] = useState(true);
-  const [isChecked, setIsChecked] = useState([]);
   const [isAscending, setIsAscending] = useState(true);
 
+  const checkTodo = useCallback(async (id, done) => {
+    const idx = todoList.findIndex(({ id: searched }) => searched === id);
+    const response = await api.checkTodo(tableName, id, done);
+    if (response) {
+      const list = [...todoList];
+      const todo = list[idx];
+      list[idx] = { title: todo.title, id, createdTime: todo.createdTime, isChecked: done };
+      setTodoList(list);
```

Comment:
Because of the "pessimistic rendering" idea we've been following in this project, we should technically be trusting the data returned FROM the api response over the data we sent TO the response. 

It's not necessarily making a difference here, but the backend is the source of truth:

Reply:
```suggestion
     const nextTodoList = [...todoList];
     // when you want to ignore the `fields` object and destructure its contents even one level further, we can nest destructuring. Though makes things less readable in some cases and we have to be careful
      const {id, createdTime, fields: {title, done}} = response;

      nextTodoList[idx] = {
        title,
        id,
        createdTime,
        isChecked: done
      };

      setTodoList(nextTodoList);
```

Reply:
Hi Darga, I had to revert is as the destructuring caused problems with the API. It might take me a while to fix this bug so I'll have to come back to it later.

Reply:
no problem, thank you for looking!